taichi/python/export_lang.cpp

```diff
@@ -979,6 +983,27 @@ void export_lang(py::module &m) {
                 "SparseMatrix only supports CPU for now.");
     return SparseMatrix(n, m);
   });
+
+  py::class_<SparseLUSolver>(m, "SparseLUSolver")
+      .def(py::init<>())
+      .def("compute", &SparseLUSolver::compute)
+      .def("analyze_pattern", &SparseLUSolver::analyze_pattern)
+      .def("factorize", &SparseLUSolver::factorize)
+      .def("solve", &SparseLUSolver::solve);
+
+  py::class_<SparseLDLTSolver>(m, "SparseLDLTSolver")
+      .def(py::init<>())
+      .def("compute", &SparseLDLTSolver::compute)
+      .def("analyze_pattern", &SparseLDLTSolver::analyze_pattern)
+      .def("factorize", &SparseLDLTSolver::factorize)
+      .def("solve", &SparseLDLTSolver::solve);
+
+  py::class_<SparseLLTSolver>(m, "SparseLLTSolver")
+      .def(py::init<>())
+      .def("compute", &SparseLLTSolver::compute)
+      .def("analyze_pattern", &SparseLLTSolver::analyze_pattern)
+      .def("factorize", &SparseLLTSolver::factorize)
+      .def("solve", &SparseLLTSolver::solve);
```

Comment:
Use a base class `SparseSolver`, and create 3 child classes `template <typename EigenSolver> EigenSparseSolver : public SparseSolver {private: EigenSolver solver_; ...}`.

Create a function method
```C++
std::unique_ptr<SparseSolver> get_sparse_solver(const std::string &name, Arch arch, DataType dtype) {
    TI_ASSERT(arch_is_cpu(arch));
    TI_ASSERT(dtype == DataType::f32);
    if (name == "LU") {
        return std::make_unique<EigenSparseSolver<LU>>();
    }
}
```

Reply:
Or we leave this refactoring to the community (good first issue!) :-)

Reply:
I remove the repeated parts and update the sparse solver according to your advice. Thanks.